File: 88-MergeSortedArray/88-MergeSortedArray.java
// Last updated: 9/14/2025, 12:54:29 PM
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(n == 0) return;
        if(m == 0){
            int i = 0;
            for(int l : nums2){
                nums1[i] = l;
                i++;
            }
        }
        else{
            int j = m - 1; // index at nums1
            int k = n - 1; // index at nums2
            for(int i = m+n-1; i >= 0; i--){
                if(j < 0){
                    nums1[i] = nums2[k];
                    k--;
                }else if(k < 0){
                    nums1[i] = nums1[j];
                    j--;
                }else{
                    if(nums1[j] > nums2[k]){
                        nums1[i] = nums1[j];
                        j--;
                    }else
                    {
                        nums1[i] = nums2[k];
                        k--;
                    }
                }
                
            }
        }    
    
    }
}

File: 88-MergeSortedArray/88-MergeSortedArray.py
# Last updated: 9/14/2025, 12:54:26 PM
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        k = m-1
        j = n-1
        p = m+n-1
        while k >= 0 and j >= 0:
            if nums1[k]<nums2[j]:
                nums1[p]=nums2[j]
                j-=1
                p-=1
            else:
                nums1[p]=nums1[k]
                k-=1
                p-=1
        if k >= 0:
            return
        if j >= 0:
            while j >= 0:
                nums1[p]=nums2[j]
                p-=1
                j-=1
